src/utility/Display-ST7735/Display-ST7735.cpp

```diff
@@ -28,44 +28,89 @@ as well as Adafruit raw 1.8" TFT display
 #include <SPI.h>
 #include "../Image.h"
 
-
-#include "../Adafruit_ASFcore.h"
-#include "../Adafruit_ASFcore/status_codes.h"
 #include "../Adafruit_ZeroDMA.h"
-#include "../Adafruit_ZeroDMA/utility/dmac.h"
-#include "../Adafruit_ZeroDMA/utility/dma.h"
 
 namespace Gamebuino_Meta {
 
 
-Adafruit_ZeroDMA myDMA;
-status_code stat; // we'll use this to read and print out the DMA status codes
+static Adafruit_ZeroDMA tftDMA;
+#define DMA_DESC_COUNT (3)
+DmacDescriptor* tftDesc[DMA_DESC_COUNT];
+DmacDescriptor* next_desc = NULL;
 // are we done yet?
-volatile bool transfer_is_done = false;
-// If you like, a callback can be used
-void dma_callback(struct dma_resource* const resource) {
-	transfer_is_done = true;
-}
-void printStatus(status_code stat) {
-	Serial.print("Status ");
-	switch (stat) {
-	case STATUS_OK:
-		Serial.println("OK"); break;
-	case STATUS_BUSY:
-		Serial.println("BUSY"); break;
-	case STATUS_ERR_INVALID_ARG:
-		Serial.println("Invalid Arg."); break;
-	default:
-		Serial.print("Unknown 0x"); Serial.println(stat); break;
+volatile uint32_t dma_desc_free_count = DMA_DESC_COUNT;
+
+//#define ENABLE_IDLE_TOGGLE_PIN (17)
+//#define ENABLE_IRQ_TOGGLE_PIN1 (19)
+//#define ENABLE_IRQ_TOGGLE_PIN2 (16)
+
+#if defined(ENABLE_IDLE_TOGGLE_PIN)
+
+static inline void wait_for_desc_available(const uint32_t min_desc_num) {
+#if defined(ENABLE_IDLE_TOGGLE_PIN)
+	PORT->Group[0].OUTSET.reg = (1 << ENABLE_IDLE_TOGGLE_PIN);
+#endif /* ENABLE_IDLE_TOGGLE_PIN */
+	while (dma_desc_free_count < min_desc_num);
+#if defined(ENABLE_IDLE_TOGGLE_PIN)
+	PORT->Group[0].OUTCLR.reg = (1 << ENABLE_IDLE_TOGGLE_PIN);
+#endif /* ENABLE_IDLE_TOGGLE_PIN */
+}
+
+static inline void wait_for_transfers_done(void) {
+#if defined(ENABLE_IDLE_TOGGLE_PIN)
+	PORT->Group[0].OUTSET.reg = (1 << ENABLE_IDLE_TOGGLE_PIN);
+#endif /* ENABLE_IDLE_TOGGLE_PIN */
```

Comment:
Please use `//` for single-line comments, rather than `/* */`. It's OK for this PR, though, just so you know in the future.